core/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -561,9 +544,10 @@ trait OptimisticTransactionImpl extends TransactionalWrite
       setNewProtocolWithFeaturesEnabledByMetadata(newMetadataTmp)
     }
 
-
-
-    assertNoDeletionVectors(newMetadataTmp, spark)
+    DeletionVectorUtils.assertNoDeletionVectors(
+      spark,
+      newMetadataTmp,
+      newProtocol.getOrElse(protocolBeforeUpdate))
```

Comment:
`newProtocol.getOrElse(protocolBeforeUpdate)` is eq. to a method call `protocol` I think?